consensus/precommit.go

```diff
@@ -23,7 +23,9 @@ func (s *precommitState) decide() {
 	precommitQH := precommits.QuorumHash()
 	if precommitQH != nil {
 		s.logger.Debug("precommit has quorum", "precommitQH", precommitQH)
-		s.enterNewState(s.commitState)
+		if s.hasVoted {
```

Comment:
query proposal here. (don't forget to add a log)

Reply:
It already called `vote()` before and should be queried